Fixes for the furniture, string and noteblock mechanics.

rafael5gr2 opened the pull request:
This PR fixes an error that could happen if someone was in creative mode and was moving a barrier block in their inventory while targeting a furniture block from 6+ blocks afar(This is also present to the string mechanic and noteblock mechanic). [furniture mechanic error](https://youtu.be/HRG7y83hK3s), [string mechanic error](https://youtu.be/hUA4caWPhE4), [noteblock mechanic error](https://youtu.be/I9MtJmJWp9A)

This PR also fixes an issue with the furniture mechanic not clearing the added block data. This could result in the following [problem](https://youtu.be/2P2dMWjeqs8) when you were middle-clicking a barrier block that was added at a location where previously was a furniture block.

Boy0000 commented on src/main/java/io/th0rgal/oraxen/mechanics/provided/gameplay/furniture/FurnitureMechanic.java:
```diff
@@ -258,7 +258,13 @@ public boolean removeSolid(World world, BlockLocation rootBlockLocation, float o
                 }
             }
-            location.getBlock().setType(Material.AIR);
+            final Block block = location.getBlock();
+            final PersistentDataContainer blockData = new CustomBlockData(block, OraxenPlugin.get());
+            blockData.remove(FURNITURE_KEY);
+            blockData.remove(SEAT_KEY);
+            blockData.remove(ROOT_KEY);
+            blockData.remove(ORIENTATION_KEY);
+            block.setType(Material.AIR);
```
This might be an issue with farmblock and stuff aswell.
From talking with the dev of CustomBlockData 2.0.0 was meant to automatically remove this.
Reading over it seems we forgot to register this.

Could you try this and see if farmblock data is also left over?
And also register the listener for it in `OraxenPLugin`, should just be `CustomBlockData.registerListener(OraxenPlugin.get());`.
I think this might handle most of these for us

Boy0000 reviewed (changes requested):
raytrace stuff looks good

Th0rgal commented:
@rafael5gr2 what do you think about Boy suggestion?

rafael5gr2 commented:
Hello,
Sorry for not responding sooner. I was on vacation.
Also sorry for closing and reopening the PR, I messed up with git.
I revoked the fix I implemented that removes the custom block data and added this line of code 
```
CustomBlockData.registerListener(this);
```
in OraxenPlugin as you told me and it seems to fix the issue.

Boy0000 commented:
@rafael5gr2 np
did u get a chance to see if it fixes farmblocks aswell?
they also store stuff in the pdc but im unsure if that would be removed by the CBD listener

rafael5gr2 commented:
@Boy0000 I am not sure if this fixes farmblocks. I haven't tested it.
